python/pyfury/_serialization.pyx

```diff
@@ -1644,29 +1656,143 @@ cdef class CollectionSerializer(Serializer):
     cpdef int16_t get_xtype_id(self):
         return -FuryType.LIST.value
 
+    cpdef int8_t write_header(self, Buffer buffer, value):
+        cdef int8_t collect_flag = COLLECTION_DEFAULT_FLAG
+        elem_type = type(next(iter(value)))
```

Comment:
This create iterator object, maybe we could just init it as None here

Reply:
This is to extract the type of the first element and avoid checking `elem_type is None` in the loop. Since the collection might be a set, an iterator retrieves it.